mlflow/utils/autologging_utils/safety.py

```diff
@@ -538,7 +539,7 @@ def call_original(*og_args, **og_kwargs):
                         )
 
                 if original_has_been_called:
-                    return original_result
+                    return patch_fn_result if patch_fn_result is not None else original_result
```

Comment:
We need allow `safe_patch` to patch the "return value", because in `patched_predict` method, in order to attach the ID to the original `predict` return value, we have to generate a new numpy array instance (see https://github.com/mlflow/mlflow/pull/4491/files#r669693569)